example/example.cpp

```diff
@@ -0,0 +1,61 @@
+#include "osrm/json_container.hpp"
+#include "osrm/engine_config.hpp"
+#include "osrm/route_parameters.hpp"
+#include "osrm/osrm.hpp"
+
+#include <string>
+#include <functional>
+
+int main(int argc, const char *argv[])
+{
+    if (argc < 2)
+    {
+        std::cerr << "Error: Not enough arguments." << std::endl
+                  << "Run " << argv[0] << " data.osrm" << std::endl;
+    }
+    try
+    {
+        osrm::EngineConfig engine_config;
+        std::string base_path(argv[1]);
+        engine_config.server_paths["ramindex"] = base_path + ".ramIndex";
+        engine_config.server_paths["fileindex"] = base_path + ".fileIndex";
+        engine_config.server_paths["hsgrdata"] = base_path + ".hsgr";
+        engine_config.server_paths["nodesdata"] = base_path + ".nodes";
+        engine_config.server_paths["edgesdata"] = base_path + ".edges";
+        engine_config.server_paths["coredata"] = base_path + ".core";
+        engine_config.server_paths["geometries"] = base_path + ".geometry";
+        engine_config.server_paths["timestamp"] = base_path + ".timestamp";
+        engine_config.server_paths["namesdata"] = base_path + ".names";
+        engine_config.use_shared_memory = false;
+
+        osrm::OSRM routing_machine(engine_config);
+
+        osrm::RouteParameters route_parameters;
+        // route is in Berlin
+        auto start = std::make_pair(52.519930, 13.438640);
+        auto target = std::make_pair(52.513191, 13.415852);
+        route_parameters.service = "viaroute";
```

Comment:
Because `RouteParameter` is still kind of coupled to our HTTP URL grammar and needs a refactor.